### DISCUSSION TITLE
Synology: Unpack not inheriting permissions

### DISCUSSION THREAD

**Discussion by fredhen** [nzbgetcom/nzbget#266] [Q&A]
Hi everyone,

I'm having a problem with Medusa not having permission to the completed directory due to the permissions on the file not having inherited permissions. My Medusa user is sc-medusa and NZBGet is sc-nzbget (from the Synocommunity), but also tested installing it with wget with the same result.

I am making use of NZBTOSICKBEARD to send a postprocessing command to Medusa to pick up the file from the folder, process it (move) and then delete.

I can see the following in Medusa:
```
2024-05-07 19:01:47 ERROR	TORNADO :: [34a67cf] Unable to move file /volume1/downloaded/completed/FolderName/FileName.mkv to /volume2/Series/Show Name/Season 2/Show Name - S02E08.mkv: PermissionError(13, 'Permission denied')
``` 

If I manually apply permissions to folders and files to give sc-medusa permissions then Medusa is able to process it as expected.

The problem from what I can gather is that the Unpack happening with NZBGet is not inheriting the permissions causing the sc-medusa user to not have the required permissions.

From my hours of searching and playing it seems like `unrar x -ai` is the solution, but that does not seem to work for me. With that in place I have also tried UMask set to 0000 and CHMOD set to 0777 (with UMask at 000 and 1000) and the results are the same. This was first tried with the 24.0 stable version and then the 24.1-testing-20240506 version.

Part of the rabbit hole led me to a similar issue for SABnzbd [mentioned in SynoCommunity](https://github.com/SynoCommunity/spksrc/issues/4940) with a big discussion in the linked [SABnzbd pull request](https://github.com/sabnzbd/sabnzbd/pull/2401) which seems to come down to -ai again, but without any CHMOD which seems to mess up with Synology's ACLs which I think is what's happening here.

To give you an idea:

The completed folder has inherited permisisons (from the share's root as set based on [Permission Management](https://github.com/SynoCommunity/spksrc/wiki/Permission-Management) mentioned in the SynoCommunity):
![msedge_qJ8HpFjGri](https://github.com/nzbgetcom/nzbget/assets/13823183/133834cc-52bf-475d-829c-722f53e43dc7)

I can also see this with ls -l:
``` 
drwxrwxrwx+ 8 root root     4096 May  7 22:54 Downloads
drwxrwxrwx+ 3 sc-nzbget synocommunity 4096 May  7 23:24 completed
``` 

The created folder during unpack also has inherited permissions along with its own:
![image](https://github.com/nzbgetcom/nzbget/assets/13823183/91b75f10-60f2-4e3c-b2d3-4f5abd282c35)

And I can see this with ls -l which seems to lose the ACL as there's no + at the end anymore although the inherited permissions are still there:
``` 
drwxrwxrwx 2 sc-nzbget synocommunity 4096 May  7 23:24 FolderName
``` 

And the resulting file:
![image](https://github.com/nzbgetcom/nzbget/assets/13823183/fd84e132-204c-4f15-a157-ddeca8ec113e)

And I can see this with ls -l:
``` 
-rw-rw-rw- 1 sc-nzbget synocommunity 88235532 Apr 21 21:38 FileName.mkv
``` 

Can anyone help me in the right direction? The main objective here is for Medusa (sc-medusa) to be able to pick up the files and move them and delete it afterwards.

Note:
This was working perfectly (with the SynoCommunity using nzbget.net as I didn't update to the .com version yet), but I had a drive failure and decided to redo the NAS from scratch. That came from DSM6 though, so I can't say if the sc-download group that existed still worked after the DSM7 upgrade, but that's obviously no longer there, but I think it's more likely that I never 'converted' to ACLs.

No idea if any of the recent permissions issues are related. I did try the testing version, but like I mentioned there does seem to be something with the Synology ACLs here.
https://github.com/nzbgetcom/nzbget/issues/263
https://github.com/nzbgetcom/nzbget/issues/228
https://github.com/nzbgetcom/nzbget/issues/165

Thank you!

**Comment by luckedea** [comment]
@fredhen Thank you for this report and review.
There have been multiple testing releases on 0506 and the umask 0 is the critical one you need - I suspect your testing didn't have that fix.
Can you please do a clean test and install the latest testing?

**Reply by velimir1845** [reply]
Hi guys, I am having the same issue where Sonarr (both v3 and v4) fails to import from the completed folder due to permissions issue. All permissions check-out and rolling back to version 20220406-29 resolves the issue. Any ideas on how to fix permission issues for version 20240115-30? The UMASK is se to  0000

**Comment by patricktrabergooglemailcom** [comment]
Same problem here, after unrar (with using x -ai) the permission gets set to this:
![Screenshot 2024-05-21 115506](https://github.com/nzbgetcom/nzbget/assets/6088503/c6909c05-a45c-4137-9195-4a700f2b9d60)

**Comment by phnzb** [comment]
@velimir1845 @patricktrabergooglemailcom
NZBGet 24+ always fixes downloads permissions based on UMask setting. NZBGet 24.0 had a bug for UMask `0` - it didn't work. Please check:
- you have 24.1 or latest testing installed, update NZBGet if needed (Settings - SYSTEM - Check for Updates)
- UMask (Settings - SECURITY - UMask) is set to `0`
- UnrarCmd `unrar -ai` is not needed

On my system (Synology DS215j / DSM 7.1.1-42962 Update 6) Sonarr v4 + NZBGet is working correctly.

**Reply by velimir1845** [reply]
Thank you @phnzb ! I have installed the native package and it is working as expected now. It looks like the Synocommunity package is still v24.0.

**Reply by phnzb** [reply]
@velimir1845
Synocommunity package can be different than v24.0 - it's complicated - it installs most recent nzbget version (stable or testing - you can choose during installation)
After install, nzbget can be updated via internal nzbget updater (Settings - SYSTEM - Check for Updates)

P.S. NZBGet v25 will switch to C++17 and native Synology packages support can be dropped due to old GCC in Synology native toolchains. So I recommend to switch to SynoCommunity version.

**Comment by patricktrabergooglemailcom** [comment]
@phnzb 
Thanks for the quick answer. The UMask setting did the trick.

Was that changed somehow in one of the latest updates? I remember that used to work until some time ago.

**Reply by phnzb** [reply]
@patricktrabergooglemailcom
Yes, NZBGet v24+ has ability to fix downloaded files permissions based on UMask. Older versions of NZBGet (before v24) - use `-ai` switch for unrar, but 7zip unfortunately doesn't support it.
